apps/server/src/controllers/userControllers/authUsers.ts

```diff
@@ -360,3 +362,68 @@ export const resetPassword = async (req: Request, res: Response) => {
     })
   }
 }
+
+export const changePassword = async (req: RequestExt, res: Response) => {
+  const passwordRegex = passwordUtils.regex
+
+  try {
+    const { _userId: userId, ...rest } = req.body
+
+    const requestData = changePasswordSchema.strict().safeParse(rest)
+
+    if (!requestData.success) {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: requestData.error.issues,
+      })
+    }
+
+    const _data = requestData.data
+
+    if (!passwordRegex.test(_data.newPassword)) {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: passwordUtils.error,
+      })
+    }
+
+    const user = await Users.findOne({
+      where: { id: userId },
+    })
+
+    // Check if the user exists
+    if (!user) {
+      return res.status(HTTP_STATUS_CODE.NOT_FOUND).json({
+        message: 'User not found',
+      })
+    }
```

Comment:
if the person is logged in, how is it possible the person would not exist?

Reply:
Yeah, I had that in mind, and I've seen where my error came from. I didn't include the user in the `rest` object. I was using the `user` variable to find a user from the `Users` table. From what I did before, if I didn't include that `if` statement block of code, the `user` variable from the next line of code will throw an error saying that the `user` can be a null value. That was the only way I could possibly find to resolve the error from the way I wrote the code.